lib/bolt/executor.rb

```diff
@@ -16,13 +17,17 @@ class Executor
     attr_reader :noop, :transports
     attr_accessor :run_as, :plan_logging
 
-    def initialize(config = Bolt::Config.new, noop = nil)
+    def initialize(config = Bolt::Config.new, analytics = Bolt::Analytics::NoopClient.new, noop = nil)
       @config = config
+      @analytics = analytics
       @logger = Logging.logger[self]
       @plan_logging = false
 
       @transports = Bolt::TRANSPORTS.each_with_object({}) do |(key, val), coll|
-        coll[key.to_s] = Concurrent::Delay.new { val.new }
+        coll[key.to_s] = Concurrent::Delay.new do
+          report_transport(key.to_s)
```

Comment:
Turns out Orch is initialized every time. So we'll need to use something else if we only want to send it on initialization.